### PR TITLE
[MRG] Allow subplots overlay

### PR DESCRIPTION
**Pull request by chenghuzi**
image

### CODE REVIEW

File: hnn_core/viz.py [hunk 1]
```diff
@@ -233,8 +246,10 @@ def plot_dipole(dpl, tmin=None, tmax=None, ax=None, layer='agg', decim=None,
         The SciPy function :func:`~scipy.signal.decimate` is used, which
         recommends values <13. To achieve higher decimation factors, a list of
         ints can be provided. These are applied successively.
-    color : tuple of float
+    color : tuple of float | str
         RGBA value to use for plotting. By default, 'k' (black)
+    sim_name : str | None
+        Prefix of dipole labels. Used when overlaying multiple simulations.
```
Comment (jasmainak) [inline hnn_core/viz.py:252]: could we just call it `label` to adhere to matplotlib naming convention? It could default to `'average'`

**Comment by jasmainak** [comment]
Let us know when ready!

**Comment by chenghuzi** [comment]
> Let us know when ready!

The functionality is now finished. I'll add some tests soon to complete it

**Comment by chenghuzi** [comment]
@jasmainak could you help to check what's going on in circleci? I tried to build the doc locally and it works fine while circleci keeps reporting errors. Once we have this figured out I think this pr will be ready to be merged

File: hnn_core/viz.py [hunk 2]
```diff
@@ -6,9 +6,22 @@
 
 import numpy as np
 from itertools import cycle
+
+import matplotlib.colors as mc
```
Comment (jasmainak) [inline hnn_core/viz.py:10]: This import is causing the trouble. I'm not sure why it doesn't get triggered in the other examples. You should "re-run job with ssh" and then try to figure what's going on

Comment (jasmainak) [reply hnn_core/viz.py:10]: Actually, aren't matplotlib imports always nested? I think that's how we have done it in MNE in the past because of known issues. Probably it's not thread safe:

https://matplotlib.org/stable/users/faq/howto_faq.html#work-with-threads

Comment (chenghuzi) [reply hnn_core/viz.py]: Bumping the matplotlib version up to >=3.5.3 seems solved the problem

File: hnn_core/gui/_viz_manager.py [hunk 3]
```diff
@@ -149,19 +159,19 @@ def _update_ax(fig, ax, single_simulation, plot_type, plot_config):
...
-            io_buf = io.BytesIO()
-            _fig.savefig(io_buf, format='raw')
-            io_buf.seek(0)
-            img_arr = np.reshape(np.frombuffer(io_buf.getvalue(),
-                                               dtype=np.uint8),
-                                 newshape=(int(_fig.bbox.bounds[3]),
-                                           int(_fig.bbox.bounds[2]), -1))
-            io_buf.close()
-            ax.imshow(img_arr)
+            with plt.ioff():
```
Comment (jasmainak) [inline hnn_core/gui/_viz_manager.py:162]: why was this necessary?

File: hnn_core/viz.py [hunk 4]
```diff
@@ -6,9 +6,23 @@
 
 import numpy as np
 from itertools import cycle
+
+import matplotlib
```
Comment (jasmainak) [inline hnn_core/viz.py:10]: I would still nest it inside the function to avoid unexpected issues.

**Review by jasmainak — APPROVED** [review]
Other than minor nitpicks, looks good to me. I did not try but I trust that you did :)

File: hnn_core/gui/_viz_manager.py (hunk 3 above)
Comment (chenghuzi) [reply hnn_core/gui/_viz_manager.py:162]: yes otherwise there will be another figure of the network object under the config tabs like this one:

image

File: hnn_core/viz.py [hunk 5]
```diff
@@ -6,9 +6,23 @@
 
 import numpy as np
 from itertools import cycle
+
```
Comment (ntolley) [inline hnn_core/viz.py:9]: 2 blank lines doesn't throw a flake8?

File: hnn_core/viz.py [hunk 6]
```diff
@@ -6,9 +6,23 @@
...
+
+import colorsys
+
 from .externals.mne import _validate_type
 
 
+def _lighten_color(color, amount=0.5):
```
Comment (ntolley) [inline hnn_core/viz.py:16]: this is pretty clever!

**Review by ntolley — APPROVED** [review]
Looks great @chenghuzi! Just played around with the GUI a bit and everything seems to be working as desired. The color cycling and label updates are quite nice too :)

File: hnn_core/dipole.py [hunk 7]
```diff
@@ -448,6 +448,10 @@ def plot_psd(self, fmin=0, fmax=None, tmin=None, tmax=None, layer='agg',
             simulation.
         layer : str, default 'agg'
             The layer to plot. Can be one of 'agg', 'L2', and 'L5'
+        color : str or tuple or None
+            The line color of PSD
+        label : str or None
+            Line label for PSD
```
Comment (rythorpe) [inline hnn_core/dipole.py:454]: Just to maintain style consistency...
```suggestion
        color : str | tuple | None
            The line color of PSD
        label : str | None
            Line label for PSD
```

**Review by rythorpe — APPROVED** [review]
This all looks great @chenghuzi! Minus my one nitpicky suggestion, I'm ready to merge this.

File: hnn_core/viz.py (hunk 5 above)
Comment (chenghuzi) [reply hnn_core/viz.py]: yeah that's a bit weird...

File: hnn_core/gui/_viz_manager.py [hunk 8]
```diff
@@ -114,7 +114,9 @@ def _update_ax(fig, ax, single_simulation, plot_type, plot_config):
 
     elif plot_type == 'PSD':
```
[the remark this answers is not in the document]
Comment (chenghuzi) [reply hnn_core/gui/_viz_manager.py:115]: Sorry I didn't get it. Do you mean to add docstrings about types that are allowed to be overlayed?

[the remark this answers is not in the document]
Comment (ntolley) [reply hnn_core/gui/_viz_manager.py:115]: Actually I'm realizing that this isn't necessary since you disable the add plot button. I originally thought there was logic in the code here specific to overlaid/non-overlaid plots but that isn't the case.
